src/workflows-app/RunDetails.ts

```diff
@@ -0,0 +1,472 @@
+  usePollingEffect(
+    () => {
+      if (cromwellProxyState && cromwellProxyState.state && workflowId) {
+        fetchWorkflowMetadata(workflowId);
+      }
+      return Promise.resolve();
+    },
+    {
+      ms: CromwellPollInterval,
+      leading: false,
+    }
+  );
+
+  /* render */
+  const renderHeader = () => {
+    const breadcrumbPathObjects = [
+      {
+        label: 'Submission History',
+        path: 'workspace-workflows-app',
+        pathParams: { name: workspaceName, namespace: workspaceBillingProject },
+        queryParams: { tab: 'submission-history' },
+      },
+      {
+        label: `Submission ${submissionId}`,
+        path: 'workspace-workflows-app-submission-details',
+        pathParams: { name: workspaceName, namespace: workspaceBillingProject, submissionId },
+      },
+      {
+        label: workflowMetadata?.workflowName,
+      },
+    ];
+
+    return h(HeaderSection, {
+      breadcrumbPathObjects,
+      button: h(SubmitNewWorkflowButton, { name: workspaceName, namespace: workspaceBillingProject }),
+      title: 'Workflow Details',
+    });
+  };
+
+  const renderFailedState = () => {
+    return h(Fragment, [
+      span({ style: { fontStyle: 'italic', marginBottom: '1rem' } }, [
+        'Failed to load workflow data. Please refresh and try again. If the problem persists, contact Terra Support for help',
+      ]),
+    ]);
+  };
+
+  const renderLoadingState = () => {
+    if (!workspaceName || !workspaceBillingProject || !submissionId || !workflowId || !workspaceId) {
+      return div({ style: { marginTop: '125px' } }, [
+        h(Spinner, { size: 48, style: { width: '100%', justifyContent: 'center' } }),
+      ]);
+    }
+    return div({ style: { width: '100%' } }, [
+      div({ style: { padding: '1rem 2rem 2rem' } }, [renderHeader()]),
+      div({ style: { display: 'flex', justifyContent: 'space-between', padding: '1rem 2rem 1rem' } }, [
+        h(WorkflowInfoBox, {
+          name: workspaceName,
+          namespace: workspaceBillingProject,
+          submissionId,
+          workflowId,
+          workspaceId,
+          showLogModal,
+        }),
+      ]),
+      div({ style: { marginTop: '125px' } }, [
+        h(Spinner, { size: 48, style: { width: '100%', justifyContent: 'center' } }),
+      ]),
+    ]);
+  };
+
+  const renderArchivedAndDeletedState = () => {
+    return h(Fragment, [
+      div({ style: { lineHeight: '24px', marginTop: '0.5rem', ...elements.sectionHeader } }, [' Run Details Archived']),
+      div({ style: { lineHeight: '24px', marginTop: '0.5rem' } }, [
+        "This run's details have been archived. Please refer to the ",
+        h(
+          Link,
+          {
+            href: 'https://support.terra.bio/hc/en-us/articles/360060601631',
+            ...newTabLinkProps,
+          },
+          [icon('pop-out', { size: 18 }), ' Run Details Archived']
+        ),
+        ' support article for details on how to access the archive.',
```

Comment:
I think it's fine for this to be here, but FYI we do not have metadata archiving in Azure. Could remove to simplify this page or leave in to simplify the eventual reintegration with Workflow Dashboard.

Reply:
Ooh I like removing stuff!